backend/src/main/java/ai/verta/modeldb/lineage/LineageDAORdbImpl.java

```diff
@@ -30,36 +49,241 @@
+        entityType = ENTITY_TYPE_VERSIONING_BLOB;
+        break;
+      default:
+        throw new ModelDBException("Unknown entry type");
+    }
+    List list = getConnectionEntities(session, invert(connectionType), entityId, entityType);
+    for (Object entity : list) {
+      ConnectionEntity connectionEntity = (ConnectionEntity) entity;
+      List connectionEntitiesById =
+          getConnectionEntitiesByIdAndConnectionType(
+              session, connectionEntity.getId(), connectionType);
+      Map<LineageElement, ConnectionEntity> inputOrOutputInDatabase =
+          getInputOrOutput(session, connectionEntitiesById);
+      result.put(
+          connectionEntity.getId(),
+          inputOrOutputInDatabase.keySet().stream()
+              .map(lineageElement -> lineageElement.toProto(session, commitInBlobHashFunction))
+              .collect(Collectors.toList()));
+    }
+    return result;
+  }
+
+  private List getConnectionEntities(
+      Session session, int connectionType, Long entityId, int entityType) {
+    String queryString;
+    queryString =
+        "from "
+            + ConnectionEntity.class.getSimpleName()
+            + " where entityId = "
+            + entityId
+            + " and entity_type = "
+            + entityType;
+    if (connectionType != CONNECTION_TYPE_ANY) {
+      queryString += " and connectionType = " + connectionType;
+    }
+    Query query = session.createQuery(queryString);
+    return query.list();
+  }
+
+  private LineageVersioningBlobEntity getLineageVersioningBlobEntity(
+      Session session, long repositoryId, String blobSha, String blobType) {
+    String queryString;
+    Object entity;
+    queryString =
+        "from "
+            + LineageVersioningBlobEntity.class.getSimpleName()
+            + " where repositoryId = '"
+            + repositoryId
+            + "' and blobSha = '"
+            + blobSha
+            + "' and blobType = '"
+            + blobType
+            + "'";
+    entity = session.createQuery(queryString).uniqueResult();
+    return (LineageVersioningBlobEntity) entity;
+  }
+
+  private LineageExperimentRunEntity getLineageExperimentRunEntity(
+      Session session, LineageEntry lineageEntry) {
+    String queryString;
+    queryString =
+        "from "
+            + LineageExperimentRunEntity.class.getSimpleName()
+            + " where experimentRunId = '"
+            + lineageEntry.getExperimentRun()
+            + "'";
+    Query query = session.createQuery(queryString);
+    Object entity = query.uniqueResult();
+    return (LineageExperimentRunEntity) entity;
+  }
+
+  private int invert(int connectionType) throws ModelDBException {
+    switch (connectionType) {
+      case CONNECTION_TYPE_INPUT:
+        return CONNECTION_TYPE_OUTPUT;
+      case CONNECTION_TYPE_OUTPUT:
+        return CONNECTION_TYPE_INPUT;
+      default:
+        throw new ModelDBException("Unknown connection type");
+    }
```

Comment:
This makes me thing it should be an enum instead of constants.

Reply:
fixed